file_stream/file_stream.go

```diff
@@ -0,0 +1,111 @@
+package file_stream
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type FileStream struct {
+	file_name string
+	reader    *os.File
+	watcher   *fsnotify.Watcher
+
+	// Used for get command completion notifications
+	Done chan bool
+
+	// Used to abort streaming when a client disconnects, etc
+	ctx context.Context
+}
+
+//-------------------------------------------------------------------------------------------------
+func NewFileStream(file_name string, ctx context.Context) *FileStream {
```

Comment:
It is a common convention for the `context.Context` to be the first argument of a function and to return an error instead of panicing
```suggestion
func NewFileStream(ctx context.Context, filename string) (*FileStream, error) {
```